src/Serializer/AbstractItemNormalizer.php

```diff
@@ -222,13 +222,19 @@ public function denormalize(mixed $data, string $class, string $format = null, a
             return $object;
         }
 
+        $options = $this->getFactoryOptions($context);
+        $propertyNames = iterator_to_array($this->propertyNameCollectionFactory->create($resourceClass, $options));
         // Revert attributes that aren't allowed to be changed after a post-denormalize check
         foreach (array_keys($data) as $attribute) {
+            if (!\in_array($attribute, $propertyNames, true)) {
```

Comment:
```suggestion
            $attribute = $this->nameConverter ? $this->nameConverter->denormalize($attribute) : $attribute;

            if (!\in_array($attribute, $propertyNames, true)) {
```

Reply:
Shouldn't the name converter be used to convert the attribute name to object property name?